### PR TITLE
Feat/nft app

### PR DESCRIPTION
**Pull request by jonty007**

### CODE REVIEW

File: packages/nft/scripts/deploy.mjs [hunk 1]
```diff
@@ -44,10 +52,29 @@ rl.question(q, async (answer) => {
...
+      console.log("Deploying Payment contract...")
+      console.log(`export ${Payment}`)
+      const paymentHelper = new PaymentHelper(computer)
+      const paymentModSpec = await paymentHelper.deploy()
+
       console.log(`
       Deploy successful
       \x1b[2mYou can copy this module specifier\x1b[0m
-      const nftModSpec = \x1b[2m'${modSpec}'\x1b[0m
+      export const nftModSpec = \x1b[2m'${modSpec}'\x1b[0m
```
Comment (ClemensLey) [inline packages/nft/scripts/deploy.mjs:74]: We should consider using "fs.writeFile" here to write directly to the .env file (after prompting the user for their ok)

File: packages/swap/src/offer.ts [hunk 2]
```diff
@@ -4,8 +4,13 @@ import { Transaction } from '@bitcoin-computer/nakamotojs'
 const { Contract } = await import('@bitcoin-computer/lib')
 
 export class Offer extends Contract {
-  constructor(owner: string, url: string, json: string) {
-    super({ _owners: [owner], _url: url, json })
+  txHex: string
+  constructor(owner: string, url: string, txHex?: string) {
```
Comment (ClemensLey) [inline packages/swap/src/offer.ts:8]: We should remove the owner here. If no owner is specified the Bitcoin Computer uses the public key of the computer object that broadcasts the transaction.

The difference is that if the owner is passed as a parameter, it is added to the metadata and we end up storing it twice (in the metadata and the output script). If we omit it, it only occurs in the output script. That makes using the contract a little cheaper...

File: packages/swap/src/offer.ts [hunk 3]
```diff
@@ -23,17 +28,31 @@ export class OfferHelper {
     return this.mod
   }
 
-  async createOfferTx(publicKey: string, url: string, tx: Transaction) {
+  async createOfferTx(publicKey: string, url: string, tx?: Transaction) {
+    const exp = tx
+      ? `new Offer("${publicKey}", "${url}", "${tx.serialize()}")`
+      : `new Offer("${publicKey}", "${url}")`
```
Comment (ClemensLey) [inline packages/swap/src/offer.ts:34]: We do not need to create and broadcast an offer transaction that does not contain an offerTx

**Comment by ClemensLey** [comment]
Great work

File: packages/components/src/Gallery.tsx [hunk 4]
```diff
@@ -177,7 +177,7 @@ export default function WithPagination<T extends Class>(q: UserQuery<T>) {
       query["limit"] = contractsPerPage + 1
       const result = await computer.query(query)
       setIsNextAvailable(result.length > contractsPerPage)
-      setRevs(result)
+      setRevs(result.slice(0, contractsPerPage))
```
[the remark this answers is not in the document]
Comment (jonty007) [reply packages/components/src/Gallery.tsx:180]: We try to get limit+1 in a single query, this logic helps us in determining if there's one more page available. We need to slice as we don't want to show this extra smart contract.

File: packages/nft/scripts/deploy.mjs (hunk 1 above)
Comment (jonty007) [reply packages/nft/scripts/deploy.mjs:74]: The file might have some other code as well, which might get override because of this

**Comment by ClemensLey** [comment]
Great

**Pull request merged**